### ISSUE TITLE
Enabling blocking of third-party cookies causes infinite spinning on white page

### ISSUE THREAD

**Issue by valynor** [metarmask/dom-distiller-reading-mode#4]
I recently enabled chrome://flags/#enable-site-per-process (strict site isolation) as recommended by Google for Spectre/Meltdown mitigation. 

Unfortunately this seems to break the DOM Distiller Reading Mode, it just shows an empty white page with a spinner at the top. 

Is it possible to fix that? Switching off strict site isolation is not an option, unfortunately.

Version 64.0.3282.186 (Offizieller Build) (64-Bit) Win7x64
Same problem on ChromeOS 64, too.

**Comment by metarmask** [comment]
Cannot reproduce on [svtnyheter.se](https://svtnyheter.se) nor [washingtonpost.com](https//washingtonpost.com) using Chrome 65.0.3325.106 beta. Are you still able to reproduce?

**Comment by valynor** [comment]
Yes. It's not working on **any** site for me. 
I just removed&reinstalled DOM Distiller Reading Mode, no change. I can open the options of your extension and choose colour/typeface but on any website I get a blank white page with a blue spinner at the top. 

I'm on Chrome Version 65.0.3325.146 (Offizieller Build) (64-Bit) now. 
The only Chrome flags I am using are: 
chrome://flags/#reduced-referrer-granularity
chrome://flags/#enable-site-per-process

**Comment by metarmask** [comment]
Still unable to reproduce unfortunately. I used 65.0.3325.146 Stable and the same flags as you. Could you try in a new profile?

**Comment by TBambition** [comment]
same Problem here.

Version 65.0.3325.146 (Offizieller Build) (64-Bit)
**Strict site isolation** is enabled in chrome://flags

activity spinner keeps spinning, but nothing happens.

**Comment by valynor** [comment]
@metarmask 
As per your request I renamed my \user data\ directory and did a fresh login&sync on Chrome. 
The problem persists in exactly the same way as described above .. white page, blue spinner on all websites. 

I also tried switching off chrome://flags/#enable-site-per-process and this also does not fix it, so that was apparently not the cause of my problem either.

**Comment by metarmask** [comment]
Am able to reproduce by enabling blocking of third-party cookies. The issue can be solved without disabling blocking by allowing the host "chrome-extension://oiembdaoobijmdmeobkalaehgifealpl/" at chrome://settings/content/cookies.

There is [a bug in the Chromium bug tracker](https://bugs.chromium.org/p/chromium/issues/detail?id=788738) which, when resolved, will help others with this issue by granting extensions this exemption automatically. You might want to consider voting for it.

**Comment by jab** [comment]
Thanks for documenting that workaround @metarmask, just tried that and it worked like a charm. Glad I found my way to this issue!

Looks like other users are hitting this though and not finding their way here, then posting to https://chrome.google.com/webstore/detail/dom-distiller-reading-mod/oiembdaoobijmdmeobkalaehgifealpl/support that the extension is broken. Might be worth replying to them with a link here, and/or updating the description in the Chrome store to include this info.

FWIW I just starred that Chrome bug (bringing the star count up to 2!). Hope the Chrome developers can fix it before long.
